src/substrait/expression/DecimalLiteral.cpp

```diff
@@ -0,0 +1,95 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/expression/DecimalLiteral.h"
+
+#include <sstream>
+
+#include "absl/numeric/int128.h"
```

Comment:
Abseil is a pretty large dependency to pull in for int128 support.  I'd rather we just vendor `int128.h`.  However, I think we can defer that for a follow-up.  I'll try and do it myself.  Or do you expect that we will be needing many more of abseil's capabilities before we're done?

Reply:
int128, status, and time are the most interesting to me but only int128 is needed so separating it out is fine.  Strings is powerful but we have most of that in modern C++ compilers.